app/components/widgets/DatasetCardWidget.tsx

```diff
@@ -18,20 +17,6 @@ export default function DatasetCardWidget({ dataset }: DatasetCardWidgetProps) {
   );
   const isInContext = Boolean(existingLayerContext);
 
-
-  useEffect(() => {
-    if (!isInContext) {
-      addContext({
-        contextType: "layer",
-        content: dataset.dataset_name,
-        datasetId: dataset.dataset_id,
-        tileUrl: dataset.tile_url,
-        layerName: dataset.dataset_name,
-      });
-    }
-    // eslint-disable-next-line react-hooks/exhaustive-deps
-  }, []); // Run only once on mount
-
```

Comment:
The dataset context was being set whenever the dataset card was mounted, which is a very different behaviour from the other context pieces, so I move it to the appropriate context action.
This was also needed for this feature to work